src/osiris_log.erl

```diff
@@ -1669,16 +1679,21 @@ write_chunk(Chunk,
                                           tail_info =
                                               {NextOffset,
                                                {Epoch, Next, Timestamp}},
-                                          segment_size = 0}};
-        {ok, _} ->
+                                          segment_size_bytes = 0,
+                                          segment_size_chunks = 0}};
+        false ->
             State#?MODULE{mode =
                               Write#write{tail_info =
                                               {NextOffset,
                                                {Epoch, Next, Timestamp}},
                                           writers = Writers,
-                                          segment_size = SegSize + Size}}
+                                          segment_size_bytes = SegSizeBytes + Size,
+                                          segment_size_chunks = SegSizeChunks + 1}}
     end.
 
+max_segment_size_reached({ok, CurrentSizeBytes}, MaxSizeBytes, CurrentSizeChunks, MaxSizeChunks) ->
```

Comment:
I find the `{ok, _}` bit a bit weird. how about passing the Fd, the cfg and the current number of chunks and work it out in the function.

Alt update the State before this function and just pass that into the max_segment_size_reached function.